engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/impl/TextCompositor.java

```diff
@@ -250,13 +253,19 @@ private void addWordsIntoTextArea(TextArea textArea, IWordRecognizer words) {
 	 *
 	 */
 	private void addWordIntoTextArea(TextArea textArea, Word word) {
+
 		// get the word's size
 		int textLength = word.getLength();
 		int wordWidth = getWordWidth(fontInfo, word);
 		// append the letter spacing
 		wordWidth += textStyle.getLetterSpacing() * textLength;
 		int adjustWordSize = fontInfo.getItalicAdjust() + wordWidth;
-		if (textArea.hasSpace(adjustWordSize)) {
+		int hyphenWidth = 0;
+		if (word.getValue().endsWith(SHY_STRING)) {
+			hyphenWidth = getTextWidth(fontInfo, "-"); // We are using the minus for computing the hyphen size, because
```

Comment:
Suggestion to move that long comment to the Javadoc of getTextWidth??

why call getTextWidth at all if you know that it returns 0?

Reply:
The trick is that the argument for getTextWidth is actually the good old ASCII MINUS character, not the SHY character.
A font will return a width of 0 for the SHY character. However, when it's visible at the end of the line, its width cannot be 0 actually.
Why should that comment go into the Javadoc comment for getTextWidth?
I think it's better to just use additional comment lines inside the code (it's really an implementation detail).